src/processes.c

```diff
@@ -174,12 +175,14 @@ init_child_signal_handlers (void)
 }
 
 static void
-pre_fork_fun_call (struct ipc_context *ctx, void *args)
+pre_fn_call (struct ipc_context *ctx, void *args)
 {
   (void) ctx;
   (void) args;
+  kb_t main_kb = (kb_t) args;
   // in a chuld we clean up every preexisting context
   ipc_destroy_contexts (ipcc);
+  set_main_kb (main_kb);
```

Comment:
Same here: why does it need to be stored here again ? wouldn't it be enough just once at the beginning of the scan?